### ISSUE TITLE
Unable to connect to any AP. Bricked?

### ISSUE THREAD

**Issue by Dustie** [EliasKotlyar/Xiaomi-Dafang-Hacks#310]
After updating the firmware on the brand new cam I'm unable to connect to any accesspoint, with security or not. Now I'm starting to think the device doesn't work at all any more. 

The LED is solid yellow, then blue for two seconds and then stays yellow. No difference if SD card is in or not. Is also doesn't say anything when I push the reset button. Holding it for 3, 5 or 30 seconds doesn't do anything either.

Did the flash brick it?

**Comment by jmtatsch** [comment]
Does your router show the dafang device? If yes comment all stuff except wifi driver/setup and sshd and see if you can get a ssh connection.

**Comment by joshstrange** [comment]
(This only matters if you can see the device from your router)
If you can't get ssh try telnet. When I first setup mine the motor calibration stuff would freeze up and ssh was setup after those commands.

**Comment by joshstrange** [comment]
Also take a look at this issue, it sounds like yours https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/issues/291

**Comment by Dustie** [comment]
Unfortunately I don't have a log function in my wifi router or my accesspoint so neither shows anything. I can only see if I gets an IP assigned from dhcp (it does not). 

>the motor calibration stuff would freeze up
Now you mention it, it doesn't run the motor calibration.

I double-checked and Notepad++ does show the file as using (unix) LF line endings.

**Comment by joshstrange** [comment]
@Dustie Did you rename the wifi config file or edit the file in-place? When I first set mine up I forgot to rename (or copy) `wpa_supplicant.conf.dist` to `wpa_supplicant.conf`? Also can you post your wifi config (SSID and password replaced with XXXX) maybe it's got a syntax error or similar?

**Comment by Dustie** [comment]
No luck with logging. The syslog in DD-WRT shows other clients connecting but nothing from the Dafang at all.

**Comment by joshstrange** [comment]
I will say the first time I tried to flash my Dafang where you hold the reset button with just the 1 file in the SD card root I forgot to rename it to demo.bin. I would go through these instructions here https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/blob/master/hacks/install_cfw.md very carefully again. I think I followed them incorrectly the first time even with reading them carefully. If you have DDWRT you should be able to see a list of active clients connected also.

**Comment by Dustie** [comment]
I tried going through the demo.bin part again and now it does the motor calibrate dance again (with and without sd card). It is now flashing yellow instead of being solid. I can't seem to find any info on what is the correct way for the LED to behave?

In DD-WRT it does not show up in the client list. It doesn't even try to connect (nothing in syslog).

**Comment by joshstrange** [comment]
I'm sorry @Dustie I'm not sure how much more I can help. The issues that I had don't appear to be the same as yours. I had a few issues with initial flash, I forgot to rename the wifi config file, and the motor calibration was messing things up. If you get it connected to wifi but the webui still isn't coming up then you might have the same final issue.

**Comment by Dustie** [comment]
Is your yellow LED solid or flashing?

**Comment by Dustie** [comment]
@Dopeyr Thanks, that LED commands is a great way to debug!

Do I understand it correctly that the motor calibration has been disabled? So it shouldn't turn at boot if it is running the modded files? And in the demo.bin part do I keep the setup button down until it starts rotating and I unplug it or only until it flashes blue? The guide seems a bit vague.

**Comment by jmtatsch** [comment]
Yes, motor calibration has been disabled because it has issues on newer hardware revisions. If you have a current checkout it shouldn't turn at boot. If it still turns, its probably running the original firmware.

For flashing the firmware, you only need to keep the setup button pressed for a couple of seconds, long enough that the boot loader knows to start a special update path. You can unplug as soon as its moving the base.  Because this indicates that the flashing stage is done, it has rebooted already and is calibrating.

**Comment by Dustie** [comment]
I got it working..

The instructions are very vague in my opinion. What I thought was a bricked device turned out being disabled motor calibration, changed LED lighting, and no Chinese speaker. 

Connecting to an access-point seems to be part black magic and part luck with which SD card you have on hand. The same settings started working after I switched SD card, but the card had worked fine with demo.bin which makes little sense. It was an 8gb card partitioned to 1gb and formatted to fat (not fat32). A class 10 high-speed Transcend i believe.

A two part change would be a huge help ( I'm not sure I remember enough about Linux internals/scripts but I'll have a look at it):

1.) Set the LED to show what is going on (yellow flashing until x, then solid, then blue flashing, then blue, etc.)
2.) A log on the SD card - especially wifi logging (syslog?)- would be a huge help


For anyone interested I used the following network settings for wpa2 with a DD-WRT AP:
```
ctrl_interface=/var/run/wpa_supplicant
ctrl_interface_group=0
ap_scan=1

network={
        ssid="xxxxx yyyyy"
        psk="xxxxyyyyzzzz"
        scan_ssid=1
        proto=RSN
        key_mgmt=WPA-PSK
        group=CCMP TKIP
        pairwise=CCMP TKIP
        priority=5
}
```

and a static IP (found in the wiki).
